### PR TITLE
CSHARP-3799: Fix flacky AcquireConnection_should_iterate_over_all_dormant_connections test

### PR DESCRIPTION
**Pull request by BorisDog**
AcquireConnection_should_iterate_over_all_dormant_connections timings tuning

### CODE REVIEW

File: tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs [hunk 1]
```diff
@@ -119,7 +119,7 @@ public void AcquireConnection_should_iterate_over_all_dormant_connections()
             var settings = _settings.With(
                 minConnections: 0,
                 maxConnections: connectionsCount,
-                waitQueueTimeout: TimeSpan.FromMilliseconds(1),
+                waitQueueTimeout: TimeSpan.FromMilliseconds(20),
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs:122]: I'm not sure about the test idea, but the change looks safe. How often this test fails in master? Are we sure that it fixes it reliably?

**Review by DmitryLukyanov — APPROVED** [review]
One question and then LGTM

File: tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs (hunk 1 above)
Comment (BorisDog) [reply tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs:122]: Approximately I observe it once in full EG run lately. Not sure 100% re. reliability, but 20ms has a big buffer over the worst case values I've seen in EG failures.

Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs:122]: maybe try to run like 5 full patches to ensure that it's actually fixed and then we will good to go with this I guess

Comment (BorisDog) [reply tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs:122]: No signs of it in 3 full runs.

**Pull request merged**